src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerTest.php

```diff
@@ -124,6 +126,85 @@ private function getContainerWithTokenStorage($token = null)
 
         return $container;
     }
+
+    public function testJson()
+    {
+        $container = $this->getMock(ContainerInterface::class);
+        $container
+            ->expects($this->once())
+            ->method('has')
+            ->with('serializer')
+            ->will($this->returnValue(false));
+
+        $controller = new TestController();
+        $controller->setContainer($container);
+
+        $response = $controller->json(array());
+        $this->assertInstanceOf(JsonResponse::class, $response);
+        $this->assertEquals('[]', $response->getContent());
+    }
+
+    public function testJsonWithSerializer()
+    {
+        $container = $this->getMock(ContainerInterface::class);
+        $container
+            ->expects($this->once())
+            ->method('has')
+            ->with('serializer')
+            ->will($this->returnValue(true));
+
+        $serializer = $this->getMock(SerializerInterface::class);
+        $serializer
+            ->expects($this->once())
+            ->method('serialize')
+            ->with(array(), 'json', array('json_encode_options' => 15))
```

Comment:
@fabpot which side are you on? I cannot tell if your comment was written before or after mine